llvm/tools/llvm-exegesis/lib/SnippetGenerator.cpp

```diff
@@ -200,7 +209,9 @@ static void setRegisterOperandValue(const RegisterOperandAssignment &ROV,
   if (ROV.Op->isExplicit()) {
     auto &AssignedValue = IB.getValueFor(*ROV.Op);
     if (AssignedValue.isValid()) {
-      assert(AssignedValue.isReg() && AssignedValue.getReg() == ROV.Reg);
+      // TODO don't re-assign register operands which are already "locked"
+      //  by Target in corresponding InstructionTemplate
+      // assert(AssignedValue.isReg() && AssignedValue.getReg() == ROV.Reg);
```

Comment:
The TODO here should be fine, but leaving the assert in as a comment is odd. It should probably be removed with additional information added to the todo if necessary.

Reply:
Addressed